src/schoology-extractor/schoology_extractor/mapping/attendance.py

```diff
@@ -40,7 +43,11 @@ def _get_status(status_code: int) -> str:
     return switcher.get(status_code, f"Unknown status: {status_code}")
 
 
-def map_to_udm(attendance: list, section_associations: pd.DataFrame) -> pd.DataFrame:
+def map_to_udm(
+    attendance: list,
+    section_associations: pd.DataFrame,
+    sync_callback: Callable = None,

```

Comment:
Since the "additional mapping" is only for one purpose - not generic - I renamed the argument to `sync_callback`.